File: src/main/webapp/src/app/models/midi-signal.ts
export class MidiSignal {
    command: number = 0;
    channel: number = 0;
    data1: number = 0;
    data2: number = 0;

    constructor(data?: any) {
        if(!data) {
        	return;
        }

        this.command = data.command;
        this.channel = data.channel;
        this.data1 = data.data1;
        this.data2 = data.data2;
    }
}

File: src/main/java/com/ascargon/rocketshow/midi/MidiSignal.java
package com.ascargon.rocketshow.midi;

import javax.sound.midi.InvalidMidiDataException;
import javax.sound.midi.ShortMessage;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.xml.bind.annotation.XmlRootElement;
import lombok.Getter;
import lombok.Setter;

@XmlRootElement
@Getter
@Setter
public class MidiSignal {

    private int command;
    private int channel;
    private int data1;
    private int data2;

    public MidiSignal() {
    }

    public MidiSignal(int command, int channel, int data1, int data2) {
        this.command = command;
        this.channel = channel;
        this.data1 = data1;
        this.data2 = data2;
    }

    public MidiSignal(ShortMessage shortMessage) {
        command = shortMessage.getCommand();
        channel = shortMessage.getChannel();
        data1 = shortMessage.getData1();
        data2 = shortMessage.getData2();
    }

    @JsonIgnore
    public ShortMessage getShortMessage() throws InvalidMidiDataException {
        ShortMessage shortMessage = new ShortMessage();
        shortMessage.setMessage(command, channel, data1, data2);
        return shortMessage;
    }

}
